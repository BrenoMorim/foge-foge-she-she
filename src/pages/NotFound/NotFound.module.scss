@import 'styles/variaveis';

.naoEncontrado {
    text-align: center;
    justify-content: center;
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    background-color: $roxo;
    color: $beje;
    width: 80vw;
    max-width: 700px;
    padding: 2.5rem;
    margin-top: 2rem;
    border-radius: 3rem;
    
    &__botao {
        background-color: $cinza;
        border-radius: 1.25rem;
        padding: 1rem;
        color: $roxo;
        font-size: 1.5rem;
        cursor: pointer;
    }
    &__botao:hover {
        opacity: 0.8;
    }
}