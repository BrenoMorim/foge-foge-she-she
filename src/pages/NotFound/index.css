@import 'styles/_variaveis.css';

.nao-encontrado {
    text-align: center;
    justify-content: center;
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    background: var(--fundo-roxo-vidro);
    box-shadow: var(--sombra);
    border: var(--borda-roxo-bidro);
    color: var(--beje);
    width: 80vw;
    max-width: 700px;
    padding: 2.5rem;
    margin-top: 2rem;
    border-radius: 3rem;
}

.nao-encontrado__botao {
    padding: 1rem;
    margin: .5rem;
    cursor: pointer;
    font-weight: bold;
    color: var(--roxo);
    background: var(--fundo-branco-vidro);
    border-radius: 2rem;
    box-shadow: var(--sombra);
    border: var(--borda-branco-vidro);
}
.nao-encontrado__botao:hover {
    border: 2px solid var(--preto);
    opacity: 0.9;
}
.nao-encontrado__botao:active {
    transform: scale(1.1);
}