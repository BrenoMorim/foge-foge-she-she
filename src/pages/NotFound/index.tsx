import { useNavigate } from "react-router-dom";
import styles from './NotFound.module.scss';

export default function NotFound() {
    
    const navegar = useNavigate();
    function voltar() {
        navegar(-1);
    }
    return (
        <section className={styles.naoEncontrado}>
            <h1>Erro 404</h1>
            <h2>Nada foi encontrado =(</h2>
            <button className={styles.naoEncontrado__botao} onClick={voltar}>Voltar</button>
        </section>
    );
}