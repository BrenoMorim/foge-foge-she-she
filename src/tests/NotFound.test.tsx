import { fireEvent, render, screen } from "@testing-library/react";
import NotFound from "pages/NotFound";
import React from "react";

const mockNavegacao = jest.fn();
jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavegacao
    }
});

describe('A página NotFound', () => {
    test('Deve ser renderizada corretamente', () => {
        const notFound = render(<NotFound/>);
        expect(notFound).toMatchSnapshot();
    });

    test('Deve conter botão para voltar para a última página', () => {
        render(<NotFound/>);

        const botaoVoltar = screen.getByText('Voltar');
        expect(botaoVoltar).toBeEnabled();

        fireEvent.click(botaoVoltar);
        expect(mockNavegacao).toBeCalledTimes(1);
    });
});