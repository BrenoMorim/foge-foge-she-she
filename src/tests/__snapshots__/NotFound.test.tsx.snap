// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A p√°gina NotFound Deve ser renderizada corretamente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="nao-encontrado base__painel-roxo base__flex-container"
      >
        <h1
          class="base__titulo"
        >
          Erro 404
        </h1>
        <h2
          class="base__subtitulo"
        >
          Nada foi encontrado =(
        </h2>
        <button
          class="base__botao"
        >
          Voltar
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="nao-encontrado base__painel-roxo base__flex-container"
    >
      <h1
        class="base__titulo"
      >
        Erro 404
      </h1>
      <h2
        class="base__subtitulo"
      >
        Nada foi encontrado =(
      </h2>
      <button
        class="base__botao"
      >
        Voltar
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
